Source/fontconfig/FCFontAssetInstaller.m

```diff
@@ -0,0 +1,296 @@
+{
+  if (_fontPath == nil)
+    {
+      if (error != NULL)
+	{
+	  NSDictionary *userInfo = [NSDictionary dictionaryWithObject: @"Font path is nil"
+							       forKey: NSLocalizedDescriptionKey];
+	  *error = [NSError errorWithDomain: @"FCFontAssetInstallerErrorDomain"
+				       code: -4001
+				   userInfo: userInfo];
+	}
+      return NO;
+    }
+
+  NSString *filename = [_fontPath lastPathComponent];
+  NSString *destinationDir;
+
+  // Determine installation directory based on options
+  destinationDir = [self systemFontsDirectory];
+  if (destinationDir == nil)
+    {
+      destinationDir = [self userFontsDirectory];
+    }
+
+  if (destinationDir == nil)
+    {
+      if (error != NULL)
+	{
+	  NSDictionary *userInfo = [NSDictionary dictionaryWithObject: @"Cannot determine font installation directory"
+							       forKey: NSLocalizedDescriptionKey];
+	  *error = [NSError errorWithDomain: @"FCFontAssetInstallerErrorDomain"
+				       code: -4002
+				   userInfo: userInfo];
+	}
+      return NO;
+    }
+
+  // Create destination directory if needed
+  NSError *dirError = nil;
+  if (![[NSFileManager defaultManager] createDirectoryAtPath: destinationDir
+				 withIntermediateDirectories: YES
+						  attributes: nil
+						       error: &dirError])
+    {
+      if (error != NULL)
+	{
+	  NSDictionary *userInfo = [NSDictionary dictionaryWithObjectsAndKeys:
+						   @"Failed to create font installation directory", NSLocalizedDescriptionKey,
+						 [dirError localizedDescription], NSLocalizedFailureReasonErrorKey,
+						 nil];
+	  *error = [NSError errorWithDomain: @"FCFontAssetInstallerErrorDomain"
+				       code: -4003
+				   userInfo: userInfo];
+	}
+      return NO;
+    }
+
+  // Copy font to destination
+  NSString *destinationPath = [destinationDir stringByAppendingPathComponent: filename];
+  NSError *copyError = nil;
+
+  // Remove existing font file if present
+  if ([[NSFileManager defaultManager] fileExistsAtPath: destinationPath])
+    {
+      [[NSFileManager defaultManager] removeItemAtPath: destinationPath error: nil];
+    }
+
+  if (![[NSFileManager defaultManager] copyItemAtPath: _fontPath
+					       toPath: destinationPath
+						error: &copyError])
+    {
+      if (error != NULL)
+	{
+	  NSDictionary *userInfo = [NSDictionary dictionaryWithObjectsAndKeys:
+						   @"Failed to copy font to installation directory", NSLocalizedDescriptionKey,
+						 [copyError localizedDescription], NSLocalizedFailureReasonErrorKey,
+						 nil];
+	  *error = [NSError errorWithDomain: @"FCFontAssetInstallerErrorDomain"
+				       code: -4004
+				   userInfo: userInfo];
+	}
+      return NO;
+    }
+
+  // Notify system of new font (platform-specific)
+#ifdef __APPLE__
+  // On macOS, fonts are automatically detected when placed in font directories
+  NSLog(@"Font installed to: %@", destinationPath);
+#else
+  NS_DURING
+    {
+      NSTask *task = [[NSTask alloc] init];
+      [task setLaunchPath: @"/usr/bin/fc-cache"];
```

Comment:
I don't like this hard coded path to the executable. Are you sure, it will always be in that directory?

Reply:
I'm using just fc-cache now, it needs to be in the path.